src/main/java/by/exadel/internship/service/impl/UserTimeSlotServiceImpl.java

```diff
@@ -0,0 +1,117 @@
+package by.exadel.internship.service.impl;
+
+
+import by.exadel.internship.dto.time_for_call.UserTimeSlotWithUserDTO;
+import by.exadel.internship.dto.time_for_call.UserTimeSlotWithUserIdDTO;
+import by.exadel.internship.dto.time_for_call.UserTimeSlotDTO;
+import by.exadel.internship.dto.UserDTO;
+import by.exadel.internship.mapper.UserTimeSlotMapper;
+import by.exadel.internship.repository.UserTimeSlotRepository;
+import by.exadel.internship.service.UserTimeSlotService;
+import by.exadel.internship.service.UserService;
+import by.exadel.internship.util.MDCLog;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+;
+
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class UserTimeSlotServiceImpl implements UserTimeSlotService {
+
+    private final UserTimeSlotRepository userTimeSlotRepository;
+    private final UserService userService;
+    private final UserTimeSlotMapper mapper;
+
+    private static final String SIMPLE_CLASS_NAME = UserTimeSlotService.class.getSimpleName();
+
+    @Override
+    public void saveUserTime(List<UserTimeSlotDTO> userTimeSlotDTOList, UUID userId) {
+        MDCLog.putClassNameInMDC(SIMPLE_CLASS_NAME);
+        log.info("Try to save free time list for user with uuid = {} in DB", userId);
+        UserDTO userDTO = userService.getById(userId);
+        List<UserTimeSlotDTO> resultUserTimeList = new ArrayList<>();
+        userTimeSlotDTOList.forEach(timeForCallUser -> {
+            checkTime(timeForCallUser);
+            separateTime(timeForCallUser, userDTO.getInterviewTime(), resultUserTimeList);
+        });
+        userDTO.setUserTimeSlots(resultUserTimeList);
```

Comment:
add new ones to existing user time slots.

Reply:
Done